src/ng/directive/ngRepeat.js

```diff
@@ -652,6 +650,12 @@ var ngRepeatDirective = ['$parse', '$animate', '$compile', function($parse, $ani
             }
           }
           lastBlockMap = nextBlockMap;
+
+          // Clear the value property from the hashFnLocals object to prevent a reference to the last value
+          // being leaked into this ngRepeatCompile function scope
+          if (hashFnLocals) {
+            hashFnLocals[valueIdentifier] = undefined;
+          }
```

Comment:
I would consider moving this right after the `locate existing items` `for` block (since this is whare `hasFnLocals` is used.

Reply:
Yeah I like that spot better 👍